File: Java/e_learn/src/main/java/gr/aueb/elearn/chapter13/Point2D.java
package gr.aueb.elearn.chapter13;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@NoArgsConstructor
public class Point2D extends Point {
    private double y;

    public Point2D(double x, double y) {
        super(x);
        this.y = y;
    }

    public void move(double x, double y) {
        super.move(x);
        this.y += y;
    }

    @Override
    protected void reset() {
        super.reset();
        this.y = 0;
    }

    @Override
    public String toString() {
        return super.toString() + "Point2D(y=" + this.y + ")";
    }

}

File: Csharp/SebAuebAug2022/PointInherit/Point2D.cs
namespace PointInherit
{
    internal class Point2D : Point, IMovable
    {

        public int Y { get; set; }

        public override string? ToString() => $"{base.ToString} Y: {this.Y}";

        public Point2D() : base() { }

        public static Point2D GetPoint2D() => new Point2D();

        public new void Move(int amount)
        {
            base.Move(amount);
            this.Y += amount;
        }
    }
}
